app/src/org/commcare/update/UpdateHelper.java

```diff
@@ -81,11 +83,11 @@ public static UpdateHelper getNewInstance(boolean autoUpdate, UpdateProgressList
     }
 
     // Main UpdateHelper function for staging updates
-    public ResultAndError<AppInstallStatus> update(String profileRef) {
-        setupUpdate(profileRef);
+    public ResultAndError<AppInstallStatus> update(String profileRef, InstallRequestSource installRequestSource) {
+        setupUpdate(profileRef, installRequestSource);
 
         try {
-            return new ResultAndError<>(stageUpdate(profileRef));
+            return new ResultAndError<>(stageUpdate(profileRef, installRequestSource));
```

Comment:
Should we call `markSuccess` here?

Reply:
Ohh nvm, it's getting set from `UpdateTask`. Perhaps, it'd be good to move the call to `register` outside as well?

Reply:
Good point, calling both `register` and `markSuccess` inside [now](https://github.com/dimagi/commcare-android/pull/2322/commits/6f8c007476e1a49900426a2c4c43dd2ac536fdbc)